internal/slicer/slicer.go

```diff
@@ -397,3 +434,148 @@ func createFile(targetPath string, pathInfo setup.PathInfo) (*fsutil.Entry, erro
 		MakeParents: true,
 	})
 }
+
+type generateManifestsOptions struct {
+	packageInfo []*archive.PackageInfo
+	selection   []*setup.Slice
+	report      *Report
+	targetDir   string
+}
+
+func generateManifests(options *generateManifestsOptions) error {
+	manifestSlices, err := locateManifestSlices(options.selection)
+	if err != nil {
+		return err
+	}
+	if len(manifestSlices) == 0 {
+		// Nothing to do.
+		return nil
+	}
+	dbw := jsonwall.NewDBWriter(&jsonwall.DBWriterOptions{
+		Schema: manifest.Schema,
+	})
+
+	// Add packages to the manifest.
+	for _, info := range options.packageInfo {
+		err := dbw.Add(&manifest.Package{
+			Kind:    "package",
+			Name:    info.Name,
+			Version: info.Version,
+			Digest:  info.Hash,
+			Arch:    info.Arch,
+		})
+		if err != nil {
+			return err
+		}
+	}
+	// Add slices to the manifest.
```

Comment:
Some line breaks between these quite independent sections would help readability. Bonus points for moving these blocks to smaller sub-functions instead, and having the global function as a simpler dispatcher.

Reply:
Agree on the line breaks. I have tried changing all the functions to helpers and I think the code becomes more complex because we need to pass all the arguments, so effectively the function needs to get a struct of manifest.Path (or whatever), other signatures are too restrictive. At that point the subfunction only needs to do `dbw.Add(arg)` so we might as well remove the extra level of indirection.

There is one exception though with packages. If we want to record them directly we can use the helper function to make it clearer, plus packages only have one possible arg `archive.PackageInfo`.

Reply:
I misread your comment, agree that we can refactor this function to call smaller functions that contain the inner loops.